File: PRACTICA2/QytetetJava/src/modeloqytetet/TituloPropiedad.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package modeloqytetet;

/**
 *
 * @author jesus
 */
public class TituloPropiedad {
    private String nombre;
    private boolean hipotecada;
    private int precioCompra;
    private int alquilerBase;
    private float factorRevalorizacion;
    private int hipotecaBase;
    private int precioEdificar;
    private int numCasas;
    private int numHoteles;
    private Jugador propietario;

    public TituloPropiedad(String nombre, int precioCompra, int alquilerBase, float factorRevalorizacion, int hipotecaBase, int precioEdificar) {
        this.nombre = nombre;
        this.hipotecada = false;
        this.precioCompra = precioCompra;
        this.alquilerBase = alquilerBase;
        this.factorRevalorizacion = factorRevalorizacion;
        this.hipotecaBase = hipotecaBase;
        this.precioEdificar = precioEdificar;
        this.hipotecada = false;
        this.numCasas = 0;
        this.numHoteles = 0;
    }
    
    int calcularCosteCancelar()
    {
        throw new UnsupportedOperationException("Sin implementar");
    }
    
    int calcularCosteHipotecar()
    {
        throw new UnsupportedOperationException("Sin implementar");
    }
    
    int calcularImporteAlquiler()
    {
       throw new UnsupportedOperationException("Sin implementar"); 
    }
    
    int calcularPrecioVenta()
    {
        throw new UnsupportedOperationException("Sin implementar");
    }
    
    void cancelarHipoteca()
    {
        
    }
    
    void cobrarAlquiler(int coste)
    {
        
    }
    
    void edificarCasa()
    {
        
    }
    
    void edificarHotel()
    {
        
    }

    public String getNombre() {
        return nombre;
    }

    public boolean isHipotecada() {
        return hipotecada;
    }

    public int getPrecioCompra() {
        return precioCompra;
    }

    public int getAlquilerBase() {
        return alquilerBase;
    }

    public float getFactorRevalorizacion() {
        return factorRevalorizacion;
    }

    public int getHipotecaBase() {
        return hipotecaBase;
    }

    public int getPrecioEdificar() {
        return precioEdificar;
    }

    public int getNumCasas() {
        return numCasas;
    }

    public int getNumHoteles() {
        return numHoteles;
    }

    public void setHipotecada(boolean hipotecada) {
        this.hipotecada = hipotecada;
    }
    
    int hipotecar()
    {
        throw new UnsupportedOperationException("Sin implementar");
    }
    
    int pagarAlquiler()
    {
        throw new UnsupportedOperationException("Sin implementar");
    }
    
    boolean propietarioEncarcelado()
    {
        throw new UnsupportedOperationException("Sin implementar");
    }
    
    boolean tengoPropietario()
    {
        throw new UnsupportedOperationException("Sin implementar");
    }

    Jugador getPropietario() {
        return propietario;
    }

    void setPropietario(Jugador propietario) {
        this.propietario = propietario;
    }
    
    

    @Override
    public String toString() {
        return "TituloPropiedad{" + "nombre=" + nombre + ", hipotecada=" + hipotecada + ", precioCompra=" + precioCompra + ", alquilerBase=" + alquilerBase + ", factorRevalorizacion=" + factorRevalorizacion + ", hipotecaBase=" + hipotecaBase + ", precioEdificar=" + precioEdificar + ", numCasas=" + numCasas + ", numHoteles=" + numHoteles + '}' + "\n";
    }
    
    
    
    
}

File: PRACTICA2/QytetetRuby/lib/titulo_propiedad.rb
#encoding: utf-8

# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
module ModeloQytetet
  class TituloPropiedad
    
    attr_reader :nombre,:precio_compra,:alquiler_base,:factor_revalorizacion,:hipoteca_base,:precio_edificar,:hipotecada,:num_casas,:num_hoteles
    attr_writer :hipotecada
    
    def initialize(nombre,precio_compra,alquiler_base,factor_revalorizacion,hipoteca_base,precio_edificar)
      @nombre = nombre
      @precio_compra = precio_compra
      @alquiler_base = alquiler_base
      @factor_revalorizacion = factor_revalorizacion
      @hipoteca_base = hipoteca_base
      @precio_edificar = precio_edificar
      @hipotecada = false
      @num_casas = 0
      @num_hoteles = 0
      @propietario = nil
    end
    
    def calcularCosteCancelar
    
        raise NotImplementedError
    end
    
    def calcularCosteHipotecar
    
        raise NotImplementedError
    end
    
    def calcularImporteAlquiler
    
       raise NotImplementedError 
    end
    
    def calcularPrecioVenta
    
        raise NotImplementedError
    end
    
    def cancelarHipoteca
    
    end    
    
    
     def cobrarAlquiler(coste)
    
     end
    
    
    def edificarCasa
    
    end   
    
    
    def edificarHotel
    
    end   
    
    
    def to_s
      "\nNombre: #{@nombre} \n Precio_compra: #{@precio_compra} \n Alquiler_base: #{@alquiler_base} \n Factor_revalorizacion: #{@factor_revalorizacion} \n Hipoteca_base: #{@hipoteca_base} \n Precio_edificar: #{@precio_edificar} \n Hipotecada: #{@hipotecada} \n Num_casas: #{@num_casas} \n Num_hoteles: #{@num_hoteles}"
    end
  end
end